src/utils/colorManipulation.test.ts

```diff
@@ -0,0 +1,231 @@
+      // draw the bg red
+      ctx.fillStyle = "rgb(255, 0, 0)";
+      ctx.fillRect(0, 0, canvas.width, canvas.height);
+
+      // draw a green square in the center
+      ctx.fillStyle = "rgb(0, 255, 0)";
+      ctx.fillRect(
+        canvas.width / 4,
+        canvas.width / 4,
+        canvas.width / 2,
+        canvas.height / 2,
+      );
+
+      const frame = ctx.getImageData(0, 0, canvas.width, canvas.height);
+      const expected = [0, 255, 0, 255];
+      const actual = getAverageColor(
+        frame,
+        canvas.height / 4,
+        canvas.width / 4,
+        (canvas.height * 3) / 4,
+        (canvas.width * 3) / 4,
+      );
+      expect(actual).toEqual(expected);
+    });
+  });
+
+  describe("computeDivisions", () => {
+    it("return even divisions", () => {
+      const verticalDivisions = 4,
+        horizontalDivisions = 2;
+      const videoDimensions: Dimensions = { height: 100, width: 100 };
+
+      // need color to complete the type, but not being tested here
+      const expected: Array<Division> = [
+        {
+          row: 0,
+          col: 0,
+          width: 50,
+          height: 25,
+          color: new Uint8ClampedArray([255, 0, 0, 255]),
+        },
+        {
+          row: 0,
+          col: 50,
+          width: 50,
+          height: 25,
+          color: new Uint8ClampedArray([255, 0, 0, 255]),
+        },
+        {
+          row: 25,
+          col: 0,
+          width: 50,
+          height: 25,
+          color: new Uint8ClampedArray([255, 0, 0, 255]),
+        },
+        {
+          row: 25,
+          col: 50,
+          width: 50,
+          height: 25,
+          color: new Uint8ClampedArray([255, 0, 0, 255]),
+        },
+        {
+          row: 50,
+          col: 0,
+          width: 50,
+          height: 25,
+          color: new Uint8ClampedArray([255, 0, 0, 255]),
+        },
+        {
+          row: 50,
+          col: 50,
+          width: 50,
+          height: 25,
+          color: new Uint8ClampedArray([255, 0, 0, 255]),
+        },
+        {
+          row: 75,
+          col: 0,
+          width: 50,
+          height: 25,
+          color: new Uint8ClampedArray([255, 0, 0, 255]),
+        },
+        {
+          row: 75,
+          col: 50,
+          width: 50,
+          height: 25,
+          color: new Uint8ClampedArray([255, 0, 0, 255]),
+        },
+      ];
+
+      const areDivisionsEqual = (d1: Division, d2: Division) =>
+        d1.width === d2.width &&
+        d1.height === d2.height &&
+        d1.row === d2.row &&
+        d1.col === d2.col;
+
+      const frame = ctx.getImageData(0, 0, canvas.width, canvas.height);
+      const actual = computeDivisions(
+        frame,
+        videoDimensions,
+        horizontalDivisions,
+        verticalDivisions,
+      );
+      expected.forEach((expectedDiv) => {
+        expect(
+          actual.some((actualDiv) => areDivisionsEqual(actualDiv, expectedDiv)),
+        ).toBe(true);
```

Comment:
how come .toBe(true) is used? cannot use something like == true?

Reply:
`== true` would have the same effect, but since I'm using the testing framework it's easier and more consistent to continue using it